File: uppgift3.py
import threading

def find_primes(n, results):
    for num in range(2, n):
        if all(num % i != 0 for i in range(2, int(num**0.5) + 1)):
            results.append(num)

# Find primes using threads
results = []
threads = []
for i in range(4):  # Using 4 threads
    thread = threading.Thread(target=find_primes, args=(100000, results))
    threads.append(thread)
    thread.start()
for thread in threads:
    thread.join()

print(f"Python Found Primes Count: {len(results)}")

File: uppgift3.cpp
#include <iostream>
#include <vector>
#include <thread>
#include <cmath>

void find_primes(int n, std::vector<int>& results) {
    for (int num = 2; num < n; ++num) {
        bool is_prime = true;
        for (int i = 2; i <= std::sqrt(num); ++i) {
            if (num % i == 0) {
                is_prime = false;
                break;
            }
        }
        if (is_prime) {
            results.push_back(num);
        }
    }
}

int main() {
    std::vector<int> results;
    std::vector<std::thread> threads;
    for (int i = 0; i < 4; ++i) {  // Using 4 threads
        threads.emplace_back(find_primes, 100000, std::ref(results));
    }
    for (auto& thread : threads) {
        thread.join();
    }
    std::cout << "C++ Found Primes Count: " << results.size() << std::endl;
    return 0;
}
